### ISSUE TITLE
FR: constructors guarding units is meaningful

### ISSUE THREAD

**Issue by Lasica** [ZeroK-RTS/Zero-K#3402]
Right now guarding constructors are useless - maybe they repair guarded unit.

I request a feature to make guarding constructors (including circle guard) behave the same as patrolling constructors with a leash (say leash being twice the constructor's range) with the following pattern:

1. If constructor is farther than twice the constructing range from its designated point - it moves to the designated point
2. If constructor has reached its point and there is no job nearby - continue following
3. Constructor looks for jobs if it has reached its position with following priorities:
   1) if the guarding target is damaged it repairs the target
   2) if any of your unit near the target is damaged it repairs the unit
   3) if any of allied unit near the target is damaged it repairs the unit
   4) if any wreck without ressurect progress is nearby - it reclaims the wreck
   5) if any energy rich reclaimable is nearby - it reclaims it

**Comment by sprunk** [comment]
Related: #1335

**Comment by GoogleFrog** [comment]
I don't want a constructor set to guard to do anything unrelated to helping the unit it is guarding.

**Comment by sprunk** [comment]
Is the core problem the request solves that if you attack-move a con alongside a combat unit, the con will start straggling massively (wandering off to reclaim/repair random crap along the way, possibly getting stuck there if there's a large wreckfield / expensive build project)?

**Comment by Lasica** [comment]
Yes, it would be nice to tell  your constructor to be leashed to a combat unit instead tying a combat unit to a constructor (with fight command) as now is your only option.

However I'd like to keep the ability of current fight command for constructors.
